api/applications/applicationRouter.js

```diff
@@ -232,59 +234,124 @@ router.post(
+ *       '409':
+ *         description: Response from error on second call involving Auth0 with some failure related to creating / provisioning a new user
+ *         note: many different error responses are possible with many different status codes, the most common one is shown below
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               items:
+ *                 status: 409
+ *                 message:  'User already exists.'
+ *      '422':
+ *        description: Response from error on third operation involving database operation with some failure related to Postgres where insertion of new profile failed
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              items:
+ *                status: 422
+ *                message:  'insert into \"profiles\" (\"profile_id\", \"role\") values ($1, $2) returning * - null value in column \"user_id\" of relation \"profiles\" violates not-null constraint'
  */
-/*
-NOTE: FE discerns if the user is a mentor/mentee and sends back the appropriate shape: 
-mentor = {validate_status & current_company }
-mentee = {validate_status}
-this endpoint can then send the validate status to the appropriate endpoint depending on whether the current_company is present or not
-  
-authRequired, and AdminRequired imports have been left (commented out) because they will likely be needed when auth0 is implemented.
-*/
-router.post(
-  '/update-validate_status/:profile_id',
-  validateStatusRequest,
-  (req, res, next) => {
-    const isMentor = req.body.current_company;
-    const { profile_id } = req.params;
 
-    const role = isMentor ? 'mentor' : 'mentee';
+router.post('/update-validate_status/:profile_id', async (req, res, next) => {
+  const { role, email, status } = req.body;
+  const { profile_id } = req.params;
+  let user_id;
 
-    axios
-      .post(`${baseURL}/update/${role}/${profile_id}`, {
-        validate_status: req.body.validate_status,
-      })
-      .then((result) => {
-        res.send({ status: result.status, message: result.data });
-      })
-      .catch((err) => {
-        next({ status: err.status, message: err.message });
-      });
+  try {
+    await axios.post(`${baseURL}/update/${role}/${profile_id}`, {
+      validate_status: status,
+    });
+    if (status === 'rejected') {
+      res.json({ status: 200, message: `${role} rejected!` });
+      return;
+    }
+  } catch (err) {
+    next({
+      status: err.response.status,
+      message: err.response.data.detail[0].msg,
+    });
+    return;
   }
-);
+
+  try {
+    const payload = {
+      email,
+      password: passGenerator(8),
+      connection,
+    };
+    const authData = await axios.post(`${issuer}api/v2/users`, payload, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+    user_id = authData.data.identities[0].user_id;
+  } catch (err) {
+    next({
```

Comment:
another approach here could be that you use a single `try catch` block instead of doing three separate try catches.

example: 
```js
try {
 define promise1()
 define promise2()
 define promise3()
 await Promise.all(promise1, promise2, promise3)
} catch(err) {
//   define err status and err message based on err here
const status = err.response.data.status ? err.response.data.status : 422
const errorMessage = same deal as above. 
}
```

This would clean up the function nicely I think and is a typical flow I've seen in the wild. 
[See Eyal Roth's explanation here](https://stackoverflow.com/questions/50557259/can-i-use-multiple-await-in-an-async-functions-try-catch-block)

Reply:
[Here is some docs on Async Await ](https://stackoverflow.com/questions/50557259/can-i-use-multiple-await-in-an-async-functions-try-catch-block)that can help guide if you choose to refactor to one single try/catch block.

Reply:
We originally tried putting all 3 async calls into one try/catch block, but our error handling wasn't great because the shape of the error object coming from each call is significantly different. It appeared that we would have to drill into the error object with some funky conditional logic in order to determine what to pass into our next(). We decided instead to separate into 3 separate try/catch blocks so we could customize the object being passed into next() in each catch block so it passed the most useful data. I'm not sure if there is a better way to manage the error handling when the objects coming back are so different, but if there is I would definitely agree that combining all of this into a single try/catch would be much better!